web-ui/src/main/resources/catalog/views/default/module.js

```diff
@@ -261,7 +261,7 @@
             uuid: md ? md.getUuid() : null,
             type:
               link.protocol.indexOf('WMTS') > -1 ? 'wmts' :
-              (link.protocol == 'ESRI:REST' ? 'esrirest' : 'wms'),
+                ((link.protocol == 'ESRI:REST') || (link.protocol == 'ESRI REST: Map Server') ? 'esrirest' : 'wms'),
```

Comment:
Maybe we can use starts with here if we are going to support all ESRI types ?

Reply:
I was reluctant to put them all in for a few reasons: ESRI REST and subtype  ESRI REST: Map Server are the two commonly used, while other others are not common and not easy to find the sample metadata to test; the list may not be complete; and the last the list may be changing.

But if you like to add them all in, I can do it, and try my best to test some if not all those subtypes if I can find any.

Reply:
We have to check on OpenLayer side if there is specific layer type depending on the type of services. Usually we have one protocol for each type of layer.

Reply:
Based on my test, one layer does work on multiple protocols (two), and I think we can add multiple if we want. It's all about ESRI protocol afterall.

Reply:
Let me know if you like to add the following:

ESRI REST: Feature Server,
ESRI REST: Image Service,
ESRI REST: Tiled Map Service,
OGC WMS

I just searched FGP, Opendata and other portals, unfortunately I have not found out any sample metadata containing any of the above protocols.

Reply:
I am adding the new fix using " starts with" you initially suggested (sorry I misunderstood your meaning then), and hopefully we now build a generic handler for all ESRI subtypes.